modules/kubernetes-addons/karpenter/locals.tf

```diff
@@ -64,16 +69,18 @@ locals {
   )
 
   default_helm_values = [templatefile("${path.module}/values.yaml", {
-    eks_cluster_id       = var.eks_cluster_id,
-    eks_cluster_endpoint = local.eks_cluster_endpoint,
-    service_account_name = local.service_account_name,
-    operating_system     = "linux"
+    eks_cluster_id            = var.eks_cluster_id,
+    eks_cluster_endpoint      = local.eks_cluster_endpoint,
+    service_account_name      = local.service_account_name,
+    node_iam_instance_profile = var.node_iam_instance_profile,
+    operating_system          = "linux"
   })]
 
   argocd_gitops_config = {
-    enable             = true
-    serviceAccountName = local.service_account_name
-    clusterName        = var.eks_cluster_id
-    clusterEndpoint    = local.eks_cluster_endpoint
+    enable                    = true
+    serviceAccountName        = local.service_account_name
+    clusterName               = var.eks_cluster_id
+    clusterEndpoint           = local.eks_cluster_endpoint
+    node_iam_instance_profile = var.node_iam_instance_profile
```

Comment:
This needs an update to https://github.com/aws-samples/ssp-eks-add-ons

Reply:
Will send a PR for that

Reply:
https://github.com/aws-samples/ssp-eks-add-ons/pull/5